src/spikeinterface/curation/auto_merge.py

```diff
@@ -230,12 +201,24 @@ def get_potential_auto_merge(
                 "knn",
                 "quality_score",
             ]
+    if force_copy and compute_needed_extensions:
+        # To avoid erasing the extensions of the user
+        sorting_analyzer = sorting_analyzer.copy()
 
     for step in steps:
         if step in _required_extensions:
             for ext in _required_extensions[step]:
-                if not sorting_analyzer.has_extension(ext):
-                    raise ValueError(f"{step} requires {ext} extension")
+                if compute_needed_extensions:
+                    if step in _templates_needed:
+                        template_ext = sorting_analyzer.get_extension("templates")
+                        if template_ext is None:
+                            sorting_analyzer.compute(["random_spikes", "templates"], **job_kwargs)
+                    params = eval(f"{step}_kwargs")
+                    params = params.get(ext, dict())
```

Comment:
What is the expented signature to have a nested dict for the extension params ?
This is a bit strange.
For me we should do it externally and when this is done automatically then we use the default

Reply:
Ok done, now if not present default is used (and a message is displayed to warn the user)